lib/Dialect/ESI/Passes/ESILowerPorts.cpp

```diff
@@ -39,273 +34,30 @@ inline static StringRef getStringAttributeOr(Operation *op, StringRef attrName,
-        origPort, "_" + element.name.getValue(), element.type, dataPorts[idx]));
-  if (!body)
-    return {};
-  return OpBuilder::atBlockBegin(body).create<hw::StructCreateOp>(
-      origPort.loc, origStruct, elementValues);
-}
-
-// Map a data value into the new operands for an instance. If the original type
-// was a channel of a struct (and we've been asked to flatten it), break it up
-// into per-field values.
-void SignalingStandard::mapInputDataPorts(OpBuilder &b, Value unwrappedData,
-                                          SmallVectorImpl<Value> &newOperands) {
-  auto chanTy = origPort.type.dyn_cast<ChannelType>();
-  Type dataPortType = chanTy ? chanTy.getInner() : origPort.type;
-  hw::StructType origStruct =
-      chanTy ? dataPortType.dyn_cast<hw::StructType>() : hw::StructType();
-
-  if (!rewriter.shouldFlattenStructs() || !origStruct) {
-    newOperands[dataPorts[0].argNum] = unwrappedData;
-  } else {
-    auto explode = b.create<hw::StructExplodeOp>(origPort.loc, unwrappedData);
-    assert(explode->getNumResults() == dataPorts.size());
-    for (auto [dataPort, fieldValue] :
-         llvm::zip(dataPorts, explode.getResults()))
-      newOperands[dataPort.argNum] = fieldValue;
-  }
-}
-
-// Build the data ports for outputs. If the original type was a channel of a
-// struct (and we've been asked to flatten it), explode the struct to create
-// individual ports.
-void SignalingStandard::buildOutputDataPorts(Value data) {
-  auto chanTy = origPort.type.dyn_cast<ChannelType>();
-  Type dataPortType = chanTy ? chanTy.getInner() : origPort.type;
-  hw::StructType origStruct =
-      chanTy ? dataPortType.dyn_cast<hw::StructType>() : hw::StructType();
-
-  if (!rewriter.shouldFlattenStructs() || !origStruct) {
-    dataPorts.push_back({});
-    rewriter.createNewOutput(origPort, "", dataPortType, data, dataPorts[0]);
-  } else {
-    ArrayRef<hw::StructType::FieldInfo> elements = origStruct.getElements();
-    dataPorts.append(elements.size(), {});
-
-    Operation *explode = nullptr;
-    if (body)
-      explode = OpBuilder::atBlockTerminator(body).create<hw::StructExplodeOp>(
-          origPort.loc, data);
-
-    for (size_t idx = 0, e = elements.size(); idx < e; ++idx) {
-      auto field = elements[idx];
-      Value fieldValue = explode ? explode->getResult(idx) : Value();
-      rewriter.createNewOutput(origPort, "_" + field.name.getValue(),
-                               field.type, fieldValue, dataPorts[idx]);
-    }
-  }
-}
-
-// Map the data ports coming off an instance back into the original ports. If
-// the original type was a channel of a struct (and we've been asked to flatten
-// it), construct the original struct from the new ports.
-Value SignalingStandard::mapOutputDataPorts(OpBuilder &b,
-                                            ArrayRef<Backedge> newResults) {
-  auto chanTy = origPort.type.dyn_cast<ChannelType>();
-  Type dataPortType = chanTy ? chanTy.getInner() : origPort.type;
```

Comment:
I don't mind if we get rid of this functionality short term, but I suspect we're going to need it.